counsel.el

```diff
@@ -1767,7 +1767,10 @@ choose between `yes-or-no-p' and `y-or-n-p'; otherwise default to
             :caller 'counsel-find-file-move))
 
 (defun counsel-find-file-mkdir-action (_x)
-  (make-directory (expand-file-name ivy-text ivy--directory)))
+  (make-directory (expand-file-name ivy-text ivy--directory))
+  (unless (eq ivy-exit 'done)
+    (with-selected-window (active-minibuffer-window)
+      (ivy--cd (expand-file-name ivy-text ivy--directory)))))
```

Comment:
I think it's safer to check that `active-minibuffer-window` does not return `nil`:

```el
(defun counsel-find-file-mkdir-action (_x)
  (let ((dir (expand-file-name ivy-text ivy--directory))
        (win (and (not (eq ivy-exit 'done))
                  (active-minibuffer-window))))
    (make-directory dir)
    (when win (with-selected-window win (ivy--cd dir)))))
```

Please also add a docstring. Thanks.